packages/dd-trace/src/opentracing/propagation/text_map.js

```diff
@@ -109,10 +109,42 @@ class TextMapPropagator {
     }
   }
 
+  _encodeOtelBaggageKey (key) {
+    let encoded = encodeURIComponent(key)
+    encoded = encoded.replaceAll('(', '%28')
+    encoded = encoded.replaceAll(')', '%29')
+    return encoded
+  }
+
   _injectBaggageItems (spanContext, carrier) {
-    spanContext._baggageItems && Object.keys(spanContext._baggageItems).forEach(key => {
-      carrier[baggagePrefix + key] = String(spanContext._baggageItems[key])
-    })
+    if (this._config.legacyBaggageEnabled) {
+      spanContext._baggageItems && Object.keys(spanContext._baggageItems).forEach(key => {
+        carrier[baggagePrefix + key] = String(spanContext._baggageItems[key])
+      })
+    }
+    if (this._hasPropagationStyle('inject', 'baggage')) {
+      if (this._config.baggageMaxItems < 1) return
+      let baggage = ''
+      let counter = 1
+      for (const [key, value] of Object.entries(spanContext._baggageItems)) {
+        baggage += `${this._encodeOtelBaggageKey(String(key).trim())}=${encodeURIComponent(String(value).trim())},`
+        if (counter === this._config.baggageMaxItems || counter > this._config.baggageMaxItems) break
+        counter += 1
+      }
+      baggage = baggage.slice(0, baggage.length - 1)
+      let buf = Buffer.from(baggage)
+      if (buf.length > this._config.baggageMaxBytes) {
+        const originalBaggages = baggage.split(',')
+        buf = buf.subarray(0, this._config.baggageMaxBytes)
```

Comment:
I have a few notes here. While this code looks mostly correct, there's an edge case in `buf.subarray(0, this._config.baggageMaxBytes)` where we could theoretically split a multi-byte character and end up with invalid UTF-8 bytes. I imagine if this happens the `buf.toString('utf8')` below would fail.

BUT, I don't think this will actually ever happen because we should never get multi-byte characters here, since we should have encoded all of them already with `encodeURIComponent(key)` above.

So now I'm thinking, if all characters in the string are guaranteed to be 1-byte ASCII characters after encoding, do we need to `Bugger.from(baggage)` at all, or can use use split the string directly without converting to UTF-8 bytes and back?

Reply:
I think you can keep a count of bytes like the way you already keep a count of items, and follow similar logic. Something like

```js
// consider this pseudo-code 😅
let itemCounter = 0
let byteCounter = 0
let item =`${...},`

itemCounter += 1
byteCounter += item.length // all characters in item are 1 byte after uri-encoding

// stop if we reach either limit
if (itemCounter > maxItems || byteCounter > maxBytes) break

// ...